mula/scheduler/schedulers/boefje.py

```diff
@@ -379,6 +435,49 @@ def is_task_allowed_to_run(self, boefje: Plugin, ooi: OOI) -> bool:
 
         return True
 
+    def is_task_rate_limited(self, task: BoefjeTask, hit: bool = True) -> bool:
+        """Checks whether a task is rate limited.
+
+        Args:
+            task: The task to check.
+            hit: Whether to hit the rate limiter or not.
+
+        Returns:
+            True if the task is rate limited, False otherwise.
+        """
+        if (rate_limit := task.boefje.rate_limit) is None:
+            return False
+
+        try:
+            parsed_rate_limit = parse(task.boefje.rate_limit.interval)  # type: ignore[union-attr]
+            if parsed_rate_limit is None:
+                raise ValueError("Invalid rate limit interval")
+        except ValueError as exc:
+            self.logger.warning(
+                "Could not parse rate limit for boefje: %s [boefje.id=%s, organisation_id=%s, scheduler_id=%s]",
+                task.boefje.id,
+                task.boefje.id,
+                self.organisation.id,
+                self.scheduler_id,
+            )
+            raise exc
+
+        with self.ctx.rate_limiter_lock:
+            # Get the identifier for the rate limiter
+            identifier_template = rate_limit.identifier
+            environment = Environment(loader=BaseLoader())
+            identifier = environment.from_string(identifier_template).render(task=task)
```

Comment:
I feel like using Jinja here is probably too much of a good thing. Woudn't fstring be more than capable enough?

Reply:
I think the main reason was to give more flexibility to the user to construct a rate limit identifier that can use the information embedded in the task.

@ammar92 am I correct in that assumption?